python/naiveautoml/commons.py

```diff
@@ -66,18 +66,18 @@ def __init__(self,
             raise ValueError(f"task_type must be in {domains_task_type}")
         self.task_type = task_type
         self.logger = logging.getLogger('naiveautoml.evalpool' if logger_name is None else logger_name)
-        if X is None:
```

Comment:
Reason for straight up removing is that it is checked to be a valid array-like type anyway, which would fail if X is None.